### PR TITLE
Update e2e setup script and DPS tests

### PR DESCRIPTION
**Pull request by schoims**

### CODE REVIEW

File: e2e/test/E2ETests.csproj [hunk 1]
```diff
@@ -16,7 +16,6 @@
     <Compile Remove="provisioning\ProvisioningE2ETests.cs" />
     <Compile Remove="provisioning\ProvisioningServiceClientE2ETests.cs" />
     <Compile Remove="provisioning\ProvisioningCertificateValidationE2ETest.cs" />
-    <Compile Remove="provisioning\ReprovisioningE2ETests.cs" />
```
Comment (drwill-ms) [inline e2e/test/E2ETests.csproj:19]: Why remove?

File: e2e/test/config/Configuration.CommonCertificates.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+{
+    public static partial class Configuration
+    {
+        public static partial class CommonCertificates
+        {
+            public static X509Certificate2 GetRootCaCertificate()
+            {
+                const string hubCert = "X509_CHAIN_ROOT_CA_CERTIFICATE";
+                X509Certificate2 cert = GetBase64EncodedCertificate(hubCert);
+                cert.NotAfter.Should().NotBeBefore(DateTime.Now, $"The X509 cert from {hubCert} has expired.");
```
Comment (drwill-ms) [inline e2e/test/config/Configuration.CommonCertificates.cs:18]: UtcNow?

File: e2e/test/config/Configuration.Provisioning.cs [hunk 3]
```diff
@@ -18,26 +21,23 @@ public static partial class Provisioning
...
+
+            // This certificate is a part of the chain whose root has been verified by the Provisioning service.
+            // The certificates used by the group enrollment tests are signed by this intermediate certificate.
+            // Chain: Root->Intermediate1->Intermediate2
+            // Certificate: Intermediate2->deviceCert
+            public static string GetGroupEnrollmentIntermediatePfxCertificateBase64()
+            {
+                const string intermediateCert = "X509_CHAIN_INTERMEDIATE2_PFX_CERTIFICATE";
+                using X509Certificate2 cert = GetBase64EncodedCertificate(intermediateCert, CertificatePassword);
+                cert.NotAfter.Should().NotBeBefore(DateTime.Now, $"The X509 cert from {intermediateCert} has expired.");
```
Comment (drwill-ms) [inline e2e/test/config/Configuration.Provisioning.cs:34]: Same here, UtcNow?

File: e2e/test/provisioning/ReprovisioningE2ETests.cs [hunk 4]
```diff
@@ -1,553 +0,0 @@
...
-using static Microsoft.Azure.Devices.E2ETests.Provisioning.ProvisioningE2ETests;
-using static Microsoft.Azure.Devices.E2ETests.Provisioning.ProvisioningServiceClientE2ETests;
-
-namespace Microsoft.Azure.Devices.E2ETests.Provisioning
-{
-    [TestClass]
-    [TestCategory("E2E")]
-    [TestCategory("DPS")]
-    [TestCategory("LongRunning")]
-    public class ReprovisioningE2ETests : E2EMsTestBase
```
Comment (drwill-ms) [inline e2e/test/provisioning/ReprovisioningE2ETests.cs:28]: Why is this being deleted?

Comment (schoims) [reply e2e/test/provisioning/ReprovisioningE2ETests.cs:28]: It no longer exists in main. Correct me if I am wrong, I heard ReprovisioningE2ETests are testing the server side, not the SDK.

File: e2e/test/E2ETests.csproj (hunk 1 above)
Comment (schoims) [reply e2e/test/E2ETests.csproj]: same reason, it's been removed in main branch

File: e2e/test/config/Configuration.Provisioning.cs (hunk 3 above)
Comment (schoims) [reply e2e/test/config/Configuration.Provisioning.cs]: Sure

**Comment by schoims** [comment]
>

File: e2e/test/Helpers/TestDeviceCallbackHandler.cs [hunk 5]
```diff
@@ -94,6 +95,8 @@ public async Task SetTwinPropertyUpdateCallbackHandlerAsync(string expectedPropN
 
                     try
                     {
+                        var patchList = patch.Cast<KeyValuePair<string, object>>().ToList();
+                        patch.Contains(expectedPropName).Should().BeFalse($"Expected patch to contain key {expectedPropName} but contained the following: {string.Join(",", patchList)}");
```
[the remark this answers is not in the document]
Comment (schoims) [reply e2e/test/Helpers/TestDeviceCallbackHandler.cs]: > Could you highlight the command where you had to change the passed-in params?

https://github.com/Azure/azure-iot-sdk-csharp/blob/choisophia/lts-e2e-setup/e2e/test/Helpers/X509Certificate2Helper.cs#L63-L99

File: vsts/vsts.yaml [hunk 6]
```diff
@@ -244,55 +289,57 @@ jobs:
       - script: |
```
Comment (abhipsaMisra) [inline vsts/vsts.yaml:289]: I think you should update the hosted agent to be same as what we have in main -> win 2022

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**